src/main/java/org/openrewrite/java/logging/slf4j/InexpensiveSLF4JLoggers.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2025 the original author or authors.
+ * <p>
+ * Licensed under the Moderne Source Available License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://docs.moderne.io/licensing/moderne-source-available-license
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.logging.slf4j;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.search.UsesMethod;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+
+import java.util.List;
+
+public class InexpensiveSLF4JLoggers extends Recipe {
+
+    static final MethodMatcher infoMethodMatcher = new MethodMatcher("org.slf4j.Logger debug(..)");
+    static final MethodMatcher debugMethodMatcher = new MethodMatcher("org.slf4j.Logger debug(..)");
+    static final MethodMatcher traceMethodMatcher = new MethodMatcher("org.slf4j.Logger debug(..)");
+    static final MethodMatcher errorMethodMatcher = new MethodMatcher("org.slf4j.Logger debug(..)");
+    static final MethodMatcher warnMethodMatcher = new MethodMatcher("org.slf4j.Logger debug(..)");
+    static final JavaTemplate ifEnabledThenLog = JavaTemplate
```

Comment:
Try to not store JavaTemplate in static fields; that triggers some early initialization, whereas there's no need as we cache JavaTemplates that are actually used.